aisynphys/pipeline/multipatch/intrinsic.py

```diff
@@ -170,25 +170,28 @@ def job_records(self, job_ids, session):
         q = q.filter(db.Experiment.ext_id.in_(job_ids))
         return q.all()
 
+try:
+    class MPSweep(Sweep):
+        """Adapter for neuroanalysis.Recording => ipfx.Sweep
+        """
+        def __init__(self, rec, t0=0):
+            # pulses may have different start times, so we shift time values to make all pulses start at t=0
+            pri = rec['primary'].copy(t0=t0)
+            cmd = rec['command'].copy()
+            t = pri.time_values
+            v = pri.data * 1e3  # convert to mV
+            holding = [i for i in rec.stimulus.items if i.description=='holding current']
+            if len(holding) == 0:
+                # TODO: maybe log this error
+                return None
+            holding = holding[0].amplitude
+            i = (cmd.data - holding) * 1e12   # convert to pA with holding current removed
+            srate = pri.sample_rate
+            sweep_num = rec.parent.key
+            # modes 'ic' and 'vc' should be expanded
+            clamp_mode = "CurrentClamp" if rec.clamp_mode=="ic" else "VoltageClamp" 
+
+            Sweep.__init__(self, t, v, i, clamp_mode, srate, sweep_number=sweep_num)
+except ImportError:
+    pass
```

Comment:
What's this for?

Reply:
I don't quite understand it, but the optional import of Sweep wasn't working for me - it tried to resolve Sweep when defining MPSweep, and failed.

Reply:
I'm curious if you've been able to successfully import intrinsic without IPFX - if so, maybe this was a quirk of the way I was running in a debug context or something.

Reply:
Oh! Of course. I forgot Sweep came from elsewhere.